lib/api/controller/base.js

```diff
@@ -632,14 +632,50 @@ class NativeController extends BaseController {
     return null;
   }
 
+  getSearchBody (request) {
```

Comment:
Looks like a duplicate of https://github.com/kuzzleio/kuzzle/blob/master/lib/service/storage/elasticsearch.js#L95

You should get the allowed search body key list with an `ask` event and the storage service must respond to this event

Reply:
It doesn't look like a duplicate to me. The purpose of this method is to fetch the search body either in the request's body, or in its optional arguments, depending on whether it's a HTTP:GET request or not.
The responsibility of the search API is to parse the provided request and pass it to the ES service afterwards.

The lines you linked are about parsing the search body, and verifying if it's correct. And that's clearly the ES service's responsibility to me.

Reply:
Hmm I don't know what happen because initially I put this comment on a list of allowed keys in the search body